release/train_tests/benchmark/logutils.py

```diff
@@ -0,0 +1,28 @@
+import logging
+import inspect
+
+
+def log_with_context(msg: str, level: str = "INFO", logger=None, *args, **kwargs):
```

Comment:
Can inject this context with LoggingAdapter instead: https://docs.python.org/3/library/logging.html#loggeradapter-objects

something like this:

```python
class ContextLoggerAdapter(logging.LoggerAdapter):
    def process(self, msg, kwargs):
        frame = inspect.currentframe().f_back.f_back  # Go two frames back
        class_name = getattr(frame.f_locals.get("self"), "__class__", None)
        func_name = frame.f_code.co_name
        prefix = f"[{class_name.__name__}/{func_name}]" if class_name else f"[{func_name}]"
        return f"{prefix} {msg}", kwargs

logger = ContextLoggerAdapter(logger, {})
```

Reply:
Then, we can keep the regular `logger.info` calls.